### PR TITLE
[MAINTENANCE] Azure Blob Stores `TupleStoreBackend` now supports `credential` string

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
# What does this PR do? 
* Small update to the `TupleStoreBackend` to support `credential` string. 
* Chose this instead of limiting access to just the `connection_string` since the `BlobServiceClient` which we use to connect to Azure, supports all 3. 
* Updated reference Environment notebook.
* Added unittest

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/data_context/store/test_store_backends.py [hunk 1]
```diff
@@ -1204,6 +1204,49 @@ def test_TupleGCSStoreBackend():  # noqa: PLR0915
     )
 
 
+@pytest.mark.big
```
Comment (anthonyburdi) [inline tests/data_context/store/test_store_backends.py]: Non blocking - but does this test need to be marked big and not unit? It looks like relevant items are mocked.

**Review by anthonyburdi — APPROVED** [review]
LGTM! One non blocking comment.

**Review by anthonyburdi — COMMENTED** [review]
LGTM! One non blocking comment.

File: tests/data_context/store/test_store_backends.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply tests/data_context/store/test_store_backends.py]: Great catch. Re-marking before merging :)

**Pull request merged**